ocamltest/main.ml

```diff
@@ -218,23 +220,19 @@ let list_tests dir =
         end
       ) (Sys.readdir dir)
   end;
-  List.rev !res
+  if !res = [] then failed := true;
+  List.iter print_endline (List.rev !res)
```

Comment:
We are in nitpicking territory (so please feel free to do nothing) but I would rather have those domain-specific functions return data (`List.rev !res` as before), and have the data turned into effects in the main function. (This is what I teach my students: data inside, effects outside.) This would force you to define a wrapper somewhere around `main`, but it also makes it easy to get rid of the global mutable variable `failed`, which is a code smell.

Reply:
I pushed a commit in this direction.

Reply:
The result looks nice, thanks!